implementations/elixir/ockam/ockam/lib/ockam/credential/authenticator/direct/client.ex

```diff
@@ -0,0 +1,121 @@
+defmodule Ockam.Credential.Authenticator.Direct.Client.AddMemberRequest do
+  @moduledoc """
+  Request data structure for Credential Authority authenticator API add_member
+  """
+  use TypedStruct
+
+  typedstruct do
+    plugin(Ockam.TypedCBOR.Plugin)
+    field(:identity_id, String.t(), minicbor: [key: 1])
+    field(:attributes, %{String.t() => String.t()}, minicbor: [key: 2])
+  end
+end
+
+defmodule Ockam.Credential.Authenticator.Direct.Client.AttributesEntry do
+  @moduledoc """
+  Data structure representing attribute entry for Credential Authority authenticator API list_members
+  """
+  use TypedStruct
+
+  typedstruct do
+    plugin(Ockam.TypedCBOR.Plugin)
+    ## Rust encodes attribute values as a list of bytes
+    field(:attributes, %{String.t() => {:list, :integer}}, minicbor: [key: 1])
```

Comment:
Is there anywhere here or outside of this changed code that needs to ensure we convert a map that has atom keys instead of string keys?

Reply:
Can you clarify the question? The keys are not converted anywhere, attributes should always assume string keys.